tfep/io/tmbarlog.py

```diff
@@ -0,0 +1,718 @@
+    │   │   └── ...
+    │   ├── epoch-1.h5
+    │   └── ...
+    ├── eval/
+    │   ├── step-0.h5
+    │   │   ├── 0_1/
+    │   │   ├── 0_2/
+    │   │   └── ...
+    │   └── ...
+    └── metadata.json
+    ```
+
+    Each HDF5 file contains groups for each state mapping, with datasets for the
+    various quantities. The data is organized hierarchically:
+    - First level: state mapping groups (e.g., "0_1" for mapping from state 0 to state 1)
+    - Second level: datasets (potential_energy, cv_values, etc.)
+
+    Finally, a JSON file is used to store metadata about the experiment such as
+    batch and epoch sizes.
+
+    """
+
+    VERSION = '0.1'
+    METADATA_FILE_NAME = 'metadata.json'
+    INDEX_NAMES = ['trajectory_sample_index', 'dataset_sample_index']
+    MASK_NAME = '__mask'
+
+    def __init__(
+            self,
+            save_dir_path='tfep_logs',
+            data_loader=None,
+            train_subdir_name='train',
+            eval_subdir_name='eval',
+            n_states=2,
+            state_names: Optional[List[str]] = None,
+    ):
+        """Constructor.
+
+        Parameters
+        ----------
+        save_dir_path : str, optional
+            The main directory where to save the training and evaluation data.
+        data_loader : torch.utils.data.DataLoader, optional
+            The data loader used for training wrapping a :class:``tfep.io.dataset.TrajectoryDataset``.
+            This must be passed when a new logger is created as it is used to
+            determine epoch, batch, and trajectory dimensions. If ``save_dir_path``
+            points to an existing logger, then this is ignored.
+        train_subdir_name : str, optional
+            The name of the subdirectory where the training data is stored.
+        eval_subdir_name : str, optional
+            The name of the subdirectory where the evaluation data is stored.
+        n_states : int, optional
+            The number of states in the system. Default is 2.
+        state_names : List[str], optional
+            Names for each state. If None, defaults to ['state_0', 'state_1', ...].
+
+        """
+        self._save_dir_path = os.path.realpath(save_dir_path)
+        self._train_dir_path = os.path.join(save_dir_path, train_subdir_name)
+        self._eval_dir_path = os.path.join(save_dir_path, eval_subdir_name)
+        
+        # Initialize state parameters
+        self._n_states = n_states
+        self._state_names = state_names if state_names is not None else [f'state_{i}' for i in range(n_states)]
+        
+        # Generate all possible state mappings
+        self._state_mappings = []
+        for i in range(n_states):
+            for j in range(n_states):
+                if i != j:  # Exclude self-mappings
+                    self._state_mappings.append((i, j))
+        
+        # This keep track of the currently in-memory training/evaluation data.
+        self._loaded_train_idx = None
```

Comment:
Similarly here, I think this code is mostly identical to `TFEPLogger.__init__`. Can you cast this so that we call `super().__init__` and then add the extra bits (like `self._state_mappings`) after that?

Reply:
After reading more, I think there might be some differences from the TFEP logger (see main comment above).